mbed_tools_ci_scripts/spdx_report/spdx_file.py

```diff
@@ -116,8 +117,8 @@ def generate_spdx_file(self) -> File:
         source_file.type = FileType.SOURCE
         source_file.comment = determine_spdx_value(None)
         source_file.chk_sum = Algorithm("SHA1", self.sha1_check_sum)
-        source_file.conc_lics = License.from_identifier(str(determine_spdx_value(self.licence)))
+        source_file.conc_lics = License.from_identifier(str(determine_spdx_value(self.licence.identifier)))
         source_file.spdx_id = f"SPDXRef-{self.id}"
         source_file.copyright = determine_spdx_value(self.copyright)
-        source_file.add_lics(License.from_identifier(str(determine_spdx_value(self.licence))))
+        source_file.add_lics(License.from_identifier(str(determine_spdx_value(self.licence.identifier))))
```

Comment:
I'm not sure why we have two ways of setting the `License` on the `source_file`? We monkey patch `conc_lics` above, then we also have this `add_lics` method. 😕

Reply:
this is the weirdness of the SDK I have at disposal I am afraid.